app/handler/books.go

```diff
@@ -45,12 +45,15 @@ func GetBook(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	params := mux.Vars(r)
 
-	for _, book := range books {
-		if book.ID == params["id"] {
-			json.NewEncoder(w).Encode(book)
-			return
-		}
+	book, err := findOrFail(params["id"])
+
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		fmt.Printf("%s \n", err)
```

Comment:
here you are swallowing the error, might be problematic for debugging, just return the error

Reply:
When I try to return err here I get this warning

```go
err error

no result values expected
```

because this func doesn't have a return type

Reply:
How do I go about this @kevinsamoei?

Reply:
for this, you need to add a return value to your function, your function signature will be `func GetBook(w http.ResponseWriter, r *http.Request) error`